### DISCUSSION TITLE
Receiving method arguments and return value on span

### DISCUSSION THREAD

**Discussion by YonatanHunger** [open-telemetry/opentelemetry-java-instrumentation#5527] [Q&A]
Is it possible to add to a span the arguments that have entered the method and the return value?
if not, what type of extension should I use to wrap the user methods?
tnx.

**Comment by mateuszrzeszutek** [comment]
Hey @YonatanHunger ,
Are you using the `@WithSpan` annotation? If so, you can use the [`@SpanAttribute`](https://github.com/open-telemetry/opentelemetry-java/blob/main/extensions%2Fannotations%2Fsrc%2Fmain%2Fjava%2Fio%2Fopentelemetry%2Fextension%2Fannotations%2FSpanAttribute.java) annotation on the method params to capture them as span attributes.
There's no annotation that can capture the return value though - for now you can do that by calling `Span.current().setAttribute("return", returnVal);` just before returning from the method.

**Reply by YonatanHunger** [reply]
Hi @mateuszrzeszutek  thank you for the answer, I am trying to do it in automatic instrumentation.

**Comment by YonatanHunger** [comment]
I think my question will be more clear when I explain that I am trying to capture all HTTP spans. I have seen [this](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers) and I want to know how can I see all these attributes?

**Reply by mateuszrzeszutek** [reply]
The request & response header capturing is configurable, you need to provide the list of headers that you want to add as span attributes - please refer to the docs [here](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/config/common.md#capturing-http-request-and-response-headers).
Once enabled, you should be able to observe these attributes in the observability backend that you're using.

About the method params/return value: do I understand it correctly that you want to add some custom attributes to the HTTP server spans? If so, you can probably start by using the `Span.current()` method from opentelemetry-api.
